src/utils/smart_agent_factory.py

```diff
@@ -0,0 +1,61 @@
+import redis
+import yaml
+import fsspec
+from logging import Logger
+from openai import AzureOpenAI
+from azure.search.documents import SearchClient
+from azure.core.credentials import AzureKeyCredential
+from distributed_cache.cache import Cache
+from functions.search_vector_function import SearchVectorFunction
+from models.agent_configuration import AgentConfiguration, agent_configuration_from_dict
+from models.settings import Settings
+from services.history import History
+from agents.smart_agent.smart_agent import Smart_Agent
+from fsspec.implementations.local import LocalFileSystem
+
+class SmartAgentFactory:
+    @staticmethod
+    def create_smart_agent(settings: Settings, session_id: str) -> Smart_Agent:
+        with open(file=settings.smart_agent_prompt_location, mode="r", encoding="utf-8") as file:
```

Comment:
I'd like to use fs spec here. If you don't make the change, its still fine; just documenting this so I can make it later.